<meta charset="UTF-8">
<h1>svmATAC</h1>

<p>Enhancement and integration of peak signal enables accurate identification of cell type in scATAC-seq</p>

<h2>Dependencies</h2>

<ul>
<li>R 3.6</li>
<li>Python 3.7</li>
</ul>

<h2>Quick Start</h2>

<p>The pipeline of svmATAC is consist of two steps:</p>

<ul>
<li><p>1.<a href="#Pre-process">Pre-process</a>:</p>

<ul>
<li>If you are trying to reproduce the results in svmATAC paper from <strong>raw sequencing data</strong>(Can be found in preprocess/XXX/input folder), you may need this step. </li>
<li>If you are trying to reproduce the results in svmATAC paper using <strong>our provided data</strong>(Can be found in XXX-dataset/XXX/input folder), or your data are already merged to 0-1 matrix (peak-cell) and labelled, you may skip this step.</li>
</ul></li>
<li><p>2.<a href="#Training-Classification">Training-Classification</a>:</p>

<ul>
<li>The training scripts and classification scripts are stored in bin/ folder of each experiment, you can try these scripts and reproduce the results in svmATAC paper through following chapters: 
<ul>
<li><a href="#intra-dataset">intra-dataset experiment</a>
<ul>
<li>Corces2016</li>
<li>Buenrostro2018</li>
<li>10xPBMCsV1</li>
<li>10xPBMCsNextGem</li>
<li>10xPBMCsV1-labeled</li>
<li>10xPBMCsNextGem-labeled</li>
</ul></li>
<li><a href="#inter-dataset">inter-dataset experiment</a>
<ul>
<li>10xPBMCs-labeled</li>
<li>10xPBMCs-unlabeled</li>
</ul></li>
</ul></li>
<li>For each single experiment:
<ul>
<li>All the scripts are available in the bin/ folder
<ul>
<li>All scripts are numbered and users should execute one by one.</li>
<li>The intermediate temporary files are stored in tmp/ folder and you can ignore these files.</li>
</ul></li>
<li>All the input data required are available in the input/ folder</li>
<li>All the output data generated are stored in the output/ folder</li>
</ul></li>
</ul></li>
</ul>

<h2>Content</h2>

<h3>Pre-process</h3>

<p>This chapter stores the scripts for processing raw data.</p>

<ul>
<li><p>The <em><strong>Corces2016</strong></em> dataset (folder <strong>'Corces2016'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Corces2016/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Corces2016/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Corces2016/output">output</a></li>
</ul></li>
<li><p>The <em><strong>Buenrostro2018</strong></em> dataset (folder <strong>'Buenrostro2018'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Buenrostro2018/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Buenrostro2018/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/Buenrostro2018/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsV1</strong></em> dataset (folder <strong>'10xPBMCsV1'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsV1/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsV1/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsV1/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsNextGem</strong></em> dataset (folder <strong>'10xPBMCsNextGem'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsNextGem/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsNextGem/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/preprocess/10xPBMCsNextGem/output">output</a></li>
</ul></li>
</ul>

<p>This chapter stores the scripts for assigning labels to 10x PBMCs v1 and nextGem scATAC-seq data from labeled scRNA-seq data using Seurat.</p>

<ul>
<li><p>The <em><strong>scRNA-seq-5k-v3</strong></em> dataset (folder <strong>'scRNA-seq-5k-v3'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scRNA-seq-5k-v3/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scRNA-seq-5k-v3/data">data</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scRNA-seq-5k-v3/output">output</a></li>
</ul></li>
<li><p>The <em><strong>scATAC-seq-5k-v1</strong></em> dataset (folder <strong>'scATAC-seq-5k-v1'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-v1/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-v1/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-v1/output">output</a></li>
</ul></li>
<li><p>The <em><strong>scATAC-seq-5k-nextgem</strong></em> dataset (folder <strong>'scATAC-seq-5k-nextgem'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-nextgem/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-nextgem/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/10x_assign_label/scATAC-seq-5k-nextgem/output">output</a></li>
</ul></li>
</ul>

<h3>Training-Classification</h3>

<h4>intra-dataset</h4>

<p>This chapter stores the scripts for intra-dataset experiments which are described in manuscript.</p>

<ul>
<li><p>The <em><strong>Corces2016</strong></em> dataset (folder <strong>'Corces2016'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Corces2016/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Corces2016/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Corces2016/output">output</a></li>
</ul></li>
<li><p>The <em><strong>Buenrostro2018</strong></em> dataset (folder <strong>'Buenrostro2018'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Buenrostro2018/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Buenrostro2018/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/Buenrostro2018/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsV1</strong></em> dataset (folder <strong>'10xPBMCsV1'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsNextGem</strong></em> dataset (folder <strong>'10xPBMCsNextGem'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsV1-labeled</strong></em> dataset (folder <strong>'10xPBMCsV1_labeled'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1_labeled/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1_labeled/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsV1_labeled/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCsNextGem-labeled</strong></em> dataset (folder <strong>'10xPBMCsNextGem_labeled'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem_labeled/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem_labeled/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/intra-dataset/10xPBMCsNextGem_labeled/output">output</a></li>
</ul></li>
</ul>

<h4>inter-dataset</h4>

<p>This chapter stores the scripts for inter-dataset experiments which are described in manuscript.</p>

<ul>
<li><p>The <em><strong>10xPBMCs-labeled</strong></em> dataset (folder <strong>'10xPBMCs_labeled'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_labeled/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_labeled/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_labeled/output">output</a></li>
</ul></li>
<li><p>The <em><strong>10xPBMCs-unlabeled</strong></em> dataset (folder <strong>'10xPBMCs_unlabeled'</strong>)</p>

<ul>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_unlabeled/bin">bin</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_unlabeled/input">input</a></li>
<li><a href="https://github.com/mrcuizhe/svmATAC/tree/master/inter-dataset/10xPBMCs_unlabeled/output">output</a></li>
</ul></li>
</ul>
